### PR TITLE
make assets builder protected

### LINKED ISSUE #8220: Allow reusing AssetsBuilder#Action from sub-classes
### Play Version (2.5.x / etc)

2.6.11

### API (Scala / Java / Neither / Both)

Scala

### Expected Behavior

Make AssetsBuilder's [Action](https://github.com/playframework/playframework/blob/6837a774f94acf98a9248977f4699a9a81869823/framework/src/play/src/main/scala/play/api/controllers/Assets.scala#L715) member field `protected` so that it can be used in specialization classes.

I need to apply some custom logic for finding the public resources of my application depending on the way it is packaged. For this I am extending AssetsBuilder and I override `#at()` and `#versioned()` methods. At the moment I've copy/pasted `private val Action = new ActionBuilder.IgnoringBody()(Execution.trampoline)` in my class to be able to create the action. 
I think it would be useful to make this member field `protected`

### PR DESCRIPTION
**Pull request by natsumisawa**
# Pull Request Checklist

* [x] Have you read [How to write the perfect pull request](https://github.com/blog/1943-how-to-write-the-perfect-pull-request)?
* [x] Have you read through the [contributor guidelines](https://www.playframework.com/contributing)?
* [x] Have you signed the [Lightbend CLA](https://www.lightbend.com/contribute/cla)?
* [x] Have you referenced any issues you're fixing using [commit message keywords](https://help.github.com/articles/closing-issues-using-keywords/)?
* [x] Have you added copyright headers to new files?
* [x] Have you checked that both Scala and Java APIs are updated?
* [x] Have you updated the documentation for both Scala and Java sections?
* [x] Have you added tests for any changed functionality?

# Helpful things

## Fixes

Fixes #8220  

## Purpose

Make AssetsBuilder's Action member field protected so that it can be used in specialization classes.

### CODE REVIEW

**Comment by lightbend-cla-validator** [comment]
Hi @natsuonatsu,

Thank you for your contribution! We really value the time you've taken to put this together.

Before we proceed with reviewing this pull request, please sign the Lightbend Contributors License Agreement:

[http://www.lightbend.com/contribute/cla](http://www.lightbend.com/contribute/cla)

**Comment by lightbend-cla-validator** [comment]
Hi @natsuonatsu,

Thank you for your contribution! We really value the time you've taken to put this together.

Before we proceed with reviewing this pull request, please sign the Lightbend Contributors License Agreement:

[http://www.lightbend.com/contribute/cla](http://www.lightbend.com/contribute/cla)

**Comment by marcospereira** [comment]
Hi @natsumisawa, thanks for taking a look at this.

There are some other places that need to be changed in order to make this happen. For example in our docs. That is why the build is broken:

https://travis-ci.org/playframework/playframework/jobs/387436689#L1410-L1432

You can run the test for docs locally and see where it is not compiling by executing the following commands:

```bash
cd documentation
sbt test
```

Let us know if you need some assistance.

Thanks.

File: framework/src/play/src/main/scala/play/api/controllers/Assets.scala [hunk 1]
```diff
@@ -712,7 +712,7 @@ package controllers {
     import meta._
     import Assets._
 
-    private val Action = new ActionBuilder.IgnoringBody()(Execution.trampoline)
+    protected val Action: ActionBuider[Request, AnyContent] = new ActionBuilder.IgnoringBody()(Execution.trampoline)
```
Comment (mkurz) [inline framework/src/play/src/main/scala/play/api/controllers/Assets.scala]: @natsumisawa Typo: `ActionBuider` -> `ActionBuilder`. Can you please fix this? Thanks!

**Comment by marcospereira** [comment]
Backport to 2.7.x: d948829d28667d9e57640162cea3c6415538e401

**Pull request merged**